cmd/daemonset-check/run_check.go

```diff
@@ -114,8 +114,8 @@ func remove(dsName string, ctx context.Context) error {
 			return fmt.Errorf("error trying to delete daemonset: %s", err)
 		}
 		log.Infoln("Successfully requested daemonset removal.")
-	case <-time.After(checkTimeLimit):
-		return errors.New("Reached check pod timeout: " + checkTimeLimit.String() + " waiting for daemonset removal command to complete.")
+	case <-time.After(checkDeadline.Sub(now)):
```

Comment:
for each of these, you could receive the channel from a single `time.After` call and then just read from it in the case:

```go
deadlineChan := time.After(checkDeadline.Sub(now))
```

then

```go
  case <-deadlineChan:
```